### ISSUE TITLE
Feature: Allow negative charge and discharge limits

### ISSUE THREAD

**Issue by vistalba** [redpomodoro/fronius_modbus#4]
I love this new custom integration! So many thanks to the dev(s) for the effort!

![image](https://github.com/user-attachments/assets/de3aa3d7-4aa9-495b-a144-73047ddb8bdc)

Source: https://loxwiki.atlassian.net/wiki/spaces/LOXEN/pages/1316061809/Fronius+Hybrid+with+Modbus+TCP?focusedCommentId=1698594976

When I look as this page, the charge_limit and discharge_limit can be positive and negative. So the possible range should be -100 to +100 instead of 0-100.
I'm not sure if this would be the best implementation option? May we also could think about different implementation idea?

Refering to the source the behavior is like this:

> Charge Limit with positive value (0 to 100%) - Charges the battery from solar, up to the maximum rate applicable to the battery size (which is dependent on the number of modules installed),

> Charge Limit with negative value (-100% to 0) - Discharges the battery to the household needs, followed by the balance going to the grid if excess flow is present. The battery will discharge over and above the set percentage if household needs are present.

> Discharge Limit with positive value (0 to 100%) - Discharges the battery to the household (up to the maximum rate applicable to the battery size, or the household requirements - whichever is the smaller value).

> Discharge Limit with negative value (-100% to 0) - Charges the battery from the grid, up to the maximum rate applicable to the battery size.

**Comment by redpomodoro** [comment]
Nice graph describing the charging/discharging behaviour! Like grid charging I'll add another control for Grid Discharging. I think that is a bit easier for new users to make sure they get the underlying settings of the BMS right.

**Comment by vistalba** [comment]
I totally agree. I'm unsure how to force the "sliders" of the other values. 

If you e.g discharge to grid, then you've to set a negative "charge_limit" value. (e.g. -10% will discharge to grid with 1,2kW).
When you create a new slider with the name "discharge_to_grid" with range 0-100%. Then I will set it to 10%.
What do we do with the charge_limit slider? As basicly in the background it is the same value. So the range ist -100% - +100%.

Do you get what are my concerns?

**Comment by redpomodoro** [comment]
To make it easier to understand the which sliders are used in the various modes I've added a table to documentation. Hopefully this helps.

The new control has been released in v0.1.2

**Comment by vistalba** [comment]
Updated. Discharge to Grid works as expected.
I‘m just curious about the sliders changing when selecting a battery mode. So e.g. when selecting Discharge to Grid it defaults to 100%.
Unsure if this is the best default behaviour?

**Comment by redpomodoro** [comment]
Not sure either, but if you use script or automation you can set the discharge power first and then the integration will that value when the mode is changed.

**Comment by vistalba** [comment]
This is exactly what I try do with NodeRed.

The problem is for example when battery mode is auto. Then setting battery mode "discharge to grid". 
Then all the sliders get overwritten with the values.

Example Sliders when Mode = Auto:
![image](https://github.com/user-attachments/assets/c64822aa-1f5b-4b42-a254-e1e2b0258684)

Example Sliders when Mode = Discharge to Grid:
![image](https://github.com/user-attachments/assets/0abe5efd-641e-4109-862d-846f334425de)

When I set `number.grid_discharge_power` to 10% and then select the mode "discharge to grid" it gets overwritten by 100%.
This leads to the conclusion that I've to change Mode, wait until `grid_discharge_power` is set to 100% and then set it to the value you want.

**Comment by redpomodoro** [comment]
In the next release I'm changing the behavior to set the charge_limit respectively discharge_limit to maximum power (ie. 100%) when 'charging limit'/'discharging limit' modes are selected. And when grid (dis)charge is selected the options it will be set to 0. So you need to change mode first and then set controls.
